controllers/clusters/zookeeper_controller.go

```diff
@@ -225,6 +228,24 @@ func (r *ZookeeperReconciler) createDefaultSecret(ctx context.Context, zk *v1bet
 	return nil
 }
 
+func (r *ZookeeperReconciler) deleteDefaultSecret(ctx context.Context, zk *v1beta1.Zookeeper) error {
+	secret := &v1.Secret{}
+	err := r.Get(ctx, types.NamespacedName{
+		Name:      fmt.Sprintf(models.DefaultUserSecretNameTemplate, models.DefaultUserSecretPrefix, zk.Name),
+		Namespace: zk.Namespace,
+	}, secret)
+
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil
+		}
+
+		return err
+	}
+
+	return r.Delete(ctx, secret)
+}
```

Comment:
Can we implement this method as a function in helpers?

Reply:
Maybe it makes sense because there are two the same methods but for different resources. I'll look for the solution

Reply:
Created deleteDefaultUserSecret function in helpers